cloudinit/distros/__init__.py

```diff
@@ -36,7 +36,7 @@
 
 OSFAMILIES = {
     'debian': ['debian', 'ubuntu'],
-    'redhat': ['centos', 'fedora', 'rhel'],
+    'redhat': ['amazon', 'centos', 'fedora', 'rhel'],
```

Comment:
One thing we might have to look over is the the variant handling in config/cloud.cfg.tmpl  this may yield unexpected results as you being building cloud-init package on amazon linux. 

Namely config/cloud.cfg.tmpl:141  may fallback to setting distro:ubuntu

```
{% if variant in ["arch", "centos", "debian", "fedora", "freebsd", "rhel", "suse", "ubuntu"] %}
   distro: {{ variant }}
{% else %}
   # Unknown/fallback distro.
   distro: ubuntu
{% endif %}
```

Reply:
@blackboxsw : Thanks for the feedback. Hopefully, it is being addressed in PR https://github.com/canonical/cloud-init/pull/119